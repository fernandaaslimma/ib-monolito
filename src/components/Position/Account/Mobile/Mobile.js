import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  PrincipalValue,
  SecondaryValue,
  Separator,
  MobileGrid,
  GridInfo,
  Label
} from "../../../../styles/mobileGrid";

import DefaultContent from "../../../common/DefaultContent";
import Icon from "../../../common/Icon";
import { black30 } from "../../../../styles/settings";

import Card from "../../../common/Card";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import ShimmerLoading from "../../../common/ShimmerLoading";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <MobileGrid columns={2}>
            <GridInfo>
              <Label>{translate("ACCOUNT_NUMBER")}</Label>
              <ShimmerLoading />
            </GridInfo>
            <GridInfo alignRight>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <ShimmerLoading />
            </GridInfo>
          </MobileGrid>
          <Separator />
          <br />
          <MobileRow>
            <Label>{translate("AVAILABLE_BALANCE")}</Label>
            <ShimmerLoading />
          </MobileRow>
          <MobileRow>
            <Label>{translate("BLOCKED_BALANCE")}</Label>
            <ShimmerLoading />
          </MobileRow>
          <MobileRow>
            <Label>{translate("TOTAL_BALANCE")}</Label>
            <ShimmerLoading />
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildAccounts = rows => {
  return rows.map((b, i) => (
    <AssetWrapper
      key={`PositionAccountMobile${i}`}
      data-test={`PositionAccountMobile${i}`}
    >
      <AssetRow>
        <Card>
          <MobileGrid columns={2}>
            <GridInfo>
              <Label>{translate("ACCOUNT_NUMBER")}</Label>
              <SecondaryValue>{b.accountNumber}</SecondaryValue>
            </GridInfo>
            <GridInfo alignRight>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <SecondaryValue>{formatDate(b.date)}</SecondaryValue>
            </GridInfo>
          </MobileGrid>
          <Separator />
          <br />
          <MobileRow>
            <Label>{translate("AVAILABLE_BALANCE")}</Label>
            <PrincipalValue>
              {formatNumber(b.availableBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{translate("BLOCKED_BALANCE")}</Label>
            <PrincipalValue>
              {formatNumber(b.blockedBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{translate("TOTAL_BALANCE")}</Label>
            <PrincipalValue>
              {formatNumber(b.totalBalance, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

function Mobile({ accounts, isEmpty, loading }) {
  return (
    <Fragment>
      <Wrapper>
        {isEmpty && (
          <DefaultContent
            data-test="Empty_Position"
            Icon={() => <Icon type="EmptyWallet" color={black30} />}
            primaryText={translate("NO_RECORDS")}
            secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
          />
        )}
        {!isEmpty &&
          (() => {
            if (loading) {
              return buildShimmerLoadingRow();
            } else {
              return buildAccounts(accounts);
            }
          })()}
      </Wrapper>
    </Fragment>
  );
}

Mobile.defaultProps = {
  accounts: []
};

Mobile.propTypes = {
  accounts: arrayOf(
    shape({
      id: number,
      accountNumber: string,
      date: string,
      totalBalance: number,
      availableBalance: number,
      blockedBalance: number
    })
  ),
  loading: bool.isRequired
};

export default Mobile;
