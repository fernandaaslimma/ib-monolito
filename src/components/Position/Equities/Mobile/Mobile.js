import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  MobileGrid,
  GridInfo,
  InfoWrapper,
  Text,
  TopHeader,
  Info,
  Label,
  TitleHeader,
  HeaderLabel,
  HeaderText,
  HeaderInfo,
  HeaderData,
  Name
} from "../../../../styles/mobileGrid";
import DefaultContent from "../../../common/DefaultContent";
import Card from "../../../common/Card";
import { isValueNullOrUndefined } from "../../../../utils/validations/input";
import { DEFAULT_VALUE } from "../../../../utils/constants";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import ShimmerLoading from "../../../common/ShimmerLoading";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <InfoWrapper>
            <Name>
              {`${translate("NAME")}: `} <ShimmerLoading />
            </Name>
          </InfoWrapper>
          <MobileRow>
            <Info>
              <Label align="left">{translate("REFERENCE_DATE")}</Label>
              <Text align="left">
                <ShimmerLoading />
              </Text>
            </Info>
            <Info>
              <Label align="right">{`${translate("PORTFOLIO")} (%)`}</Label>
              <Text align="right">
                <ShimmerLoading />
              </Text>
            </Info>
            <Info>
              <Label align="right">{`${translate(
                "GROSS_BALANCE"
              )} (R$)`}</Label>
              <Text align="right">
                <ShimmerLoading />
              </Text>
            </Info>
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildRows = (rows, loading) => {
  if (loading) {
    return buildShimmerLoadingRow();
  }
  return rows.map((b, i) => (
    <AssetWrapper key={`mobileEquity_${i}`}>
      <AssetRow>
        <Card data-test={`Card-${i}`}>
          <InfoWrapper>
            <Name>{`${translate("NAME")}: ${b.name}`}</Name>
          </InfoWrapper>
          <MobileGrid>
            <GridInfo>
              <Label>{translate("REFERENCE_DATE")}</Label>
              <Text>{formatDate(b.date)}</Text>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("PORTFOLIO")} (%)`}</Label>
              <Text>{formatNumber(b.portfolioShare, { digits: 2 })}</Text>
            </GridInfo>
            <GridInfo>
              <Label>{`${translate("GROSS_BALANCE")} (R$)`}</Label>
              <Text>{formatNumber(b.grossBalance, { digits: 2 })}</Text>
            </GridInfo>
          </MobileGrid>
          <MobileGrid>
            <GridInfo>
              <Label>{translate("QUANTITY")}</Label>
              <Text>
                {isValueNullOrUndefined(b.quantity)
                  ? DEFAULT_VALUE
                  : formatNumber(b.quantity, { digits: 0 })}
              </Text>
            </GridInfo>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

const buildTopInfo = (totalEquities, loading) => {
  const grossBalance = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totalEquities && totalEquities.grossBalance, { digits: 2 })
  );

  return (
    <TopHeader>
      <TitleHeader>{translate("TOTAL")}</TitleHeader>
      <HeaderData>
        <HeaderInfo>
          <HeaderLabel>{`${translate("GROSS_BALANCE")} (R$)`}</HeaderLabel>
          <HeaderText>{grossBalance}</HeaderText>
        </HeaderInfo>
      </HeaderData>
    </TopHeader>
  );
};

function Mobile({ equities, totalEquities, isEmpty, loading }) {
  return (
    <Fragment>
      {isEmpty ? (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      ) : (
        <Wrapper>
          {buildTopInfo(totalEquities, loading)}
          {buildRows(equities, loading)}
        </Wrapper>
      )}
    </Fragment>
  );
}

Mobile.defaultProps = {
  equities: [],
  loading: false,
  isEmpty: false
};

Mobile.propTypes = {
  equities: arrayOf(
    shape({
      name: string,
      date: string,
      issuer: string,
      maturityDate: string,
      grossBalance: number,
      netBalance: number,
      quantity: number,
      portfolioShare: number,
      indexerRate: number,
      indexer: string,
      incomeTaxBalance: number,
      iofBalance: number
    })
  ),
  loading: bool,
  isEmpty: bool
};

export default Mobile;
