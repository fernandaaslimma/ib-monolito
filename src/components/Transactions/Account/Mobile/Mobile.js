import React from "react";
import { arrayOf, shape, bool, string, array, object } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  PrincipalValue,
  SecondaryValue,
  Separator,
  MobileGrid,
  GridInfo,
  GridInfoLeftBorder,
  Label
} from "../../../../styles/mobileGrid";
import { BalanceAmountRow, Title } from "../../styles";

import Card from "../../../common/Card";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import ShimmerLoading from "../../../common/ShimmerLoading";

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <MobileRow>
            <Label>{`${translate("OPEN_BALANCE_ON")}:`}</Label>
            <ShimmerLoading />
          </MobileRow>

          <Separator noMargin />
          <MobileGrid>
            <GridInfo>
              <Label>{translate("DATE")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfoLeftBorder>
              <Label> {translate("TYPE")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfoLeftBorder>
            <GridInfoLeftBorder>
              <Label> {`${translate("VALUE")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfoLeftBorder>
          </MobileGrid>
          <Separator noMargin />

          <MobileRow>
            <Label>{`${translate("CLOSE_BALANCE_ON")}:`}</Label>
            <ShimmerLoading />
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildTransactions = rows => {
  return rows.map((b, i) => (
    <MobileGrid key={`MobileGrid_${i}`} data-test={`MobileGrid-${i}`}>
      <GridInfo>
        <Label>{translate("DATE")}</Label>
        <SecondaryValue>{formatDate(b.date)}</SecondaryValue>
      </GridInfo>
      <GridInfoLeftBorder>
        <Label>{translate("TYPE")}</Label>
        <SecondaryValue>{b.type}</SecondaryValue>
      </GridInfoLeftBorder>
      <GridInfoLeftBorder>
        <Label>{`${translate("VALUE")} (R$)`}</Label>
        <SecondaryValue>
          {formatNumber(b.amount, { digits: 2 })}{" "}
        </SecondaryValue>
      </GridInfoLeftBorder>
    </MobileGrid>
  ));
};

const buildAccounts = accounts => {
  return accounts.map((account, index) => (
    <AssetWrapper key={index}>
      <AssetRow>
        <Title>{`${translate("CHECKING_ACCOUNT")} ${
          account.userCashAccount
        }`}</Title>

        <Card>
          <MobileRow>
            <Label>{`${translate("OPEN_BALANCE_ON")} ${formatDate(
              account.accountOpenBalance.date
            )}`}</Label>
            <PrincipalValue>
              <BalanceAmountRow
                red={account.accountOpenBalance.openBalance < 0}
              >
                {formatNumber(account.accountOpenBalance.openBalance, {
                  digits: 2
                })}
              </BalanceAmountRow>
            </PrincipalValue>
          </MobileRow>
          <Separator noMargin />

          {buildTransactions(account.cashAccountTransactions)}

          <Separator noMargin />
          <br />
          <MobileRow>
            <Label>{`${translate("CLOSE_BALANCE_ON")} ${formatDate(
              account.accountCloseBalance.date
            )}`}</Label>
            <PrincipalValue>
              <BalanceAmountRow
                red={account.accountCloseBalance.closeBalance < 0}
              >
                {formatNumber(account.accountCloseBalance.closeBalance, {
                  digits: 2
                })}
              </BalanceAmountRow>
            </PrincipalValue>
          </MobileRow>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

function Mobile({ cashAccounts, loading }) {
  return (
    <Wrapper>
      {loading && buildShimmerLoadingRow()}
      {!loading && buildAccounts(cashAccounts)};
    </Wrapper>
  );
}

Mobile.defaultProps = {
  cashAccounts: [],
  loading: true
};

Mobile.propTypes = {
  loading: bool,
  cashAccounts: arrayOf(
    shape({
      userCashAccount: string,
      cashAccountTransactions: array,
      accountOpenBalance: object,
      accountCloseBalance: object
    })
  )
};

export default Mobile;
