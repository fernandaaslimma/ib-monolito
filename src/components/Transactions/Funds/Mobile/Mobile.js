import React, { Fragment } from "react";
import { arrayOf, shape, number, string, bool, func } from "prop-types";

import { translate } from "../../../../utils/i18n";
import {
  Wrapper,
  LinkCardWrapper,
  AssetWrapper,
  AssetRow,
  MobileRow,
  InfoWrapper,
  IncomeName,
  PrincipalValue,
  SecondaryValue,
  Separator,
  MobileGrid,
  GridInfo,
  GridInfoLeftBorder,
  Label,
  TopHeader,
  TitleHeader,
  HeaderData,
  HeaderInfo,
  HeaderLabel,
  HeaderText,
  Name
} from "../../../../styles/mobileGrid";

import loadable from "loadable-components";
import Pagination from "../../../common/Table/Pagination";

import DefaultContent from "../../../common/DefaultContent";
import Card from "../../../common/Card";
import formatNumber from "../../../../utils/formatNumber";
import formatDate from "../../../../utils/formatDate";
import { black30 } from "../../../../styles/settings";
import Icon from "../../../common/Icon";
import ShimmerLoading from "../../../common/ShimmerLoading";
import LinkCard from "../../../common/LinkCard";

const Filter = loadable(() => import("../../../common/Table/Filter"));
const PAGE_SIZE = 10;

const generateSum = transactions => {
  const initialTransactions = {
    grossValue: 0,
    netValue: 0,
    incomeTax: 0,
    iof: 0
  };

  return transactions.reduce((acc, current) => {
    acc.grossValue += current.grossValue;
    acc.netValue += current.netValue;
    acc.incomeTax += current.incomeTax;
    acc.iof += current.iof;
    return acc;
  }, initialTransactions);
};

const buildShimmerLoadingRow = () => {
  return (
    <AssetWrapper>
      <AssetRow>
        <Card>
          <MobileRow>
            <IncomeName>
              <ShimmerLoading />
            </IncomeName>
            <InfoWrapper>
              <Label>{`${translate("DATE")}:`}</Label>
              <ShimmerLoading />
            </InfoWrapper>
          </MobileRow>
          <MobileRow>
            <InfoWrapper>
              <Name>
                {`${translate("GROSS_VALUE")}: `}
                <ShimmerLoading />
              </Name>
            </InfoWrapper>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("NET_VALUE")}`}</Label>
            <PrincipalValue>
              <ShimmerLoading />
            </PrincipalValue>
          </MobileRow>
          <Separator />
          <MobileGrid>
            <GridInfo>
              <Label>{translate("APPL_DATE")}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfo>
            <GridInfoLeftBorder>
              <Label> {`${translate("INCOME_TAX")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfoLeftBorder>
            <GridInfoLeftBorder>
              <Label> {`${translate("IOF")} (R$)`}</Label>
              <SecondaryValue>
                <ShimmerLoading />
              </SecondaryValue>
            </GridInfoLeftBorder>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  );
};

const buildRows = (rows, loading) => {
  if (loading) {
    return buildShimmerLoadingRow();
  }
  return rows.map((b, i) => (
    <AssetWrapper key={`mobileFunds_${i}`}>
      <AssetRow>
        <Card data-test={`Card-${i}`}>
          <MobileRow>
            <InfoWrapper>
              <Name>{b.assetName}</Name>
            </InfoWrapper>
            <InfoWrapper>
              <Name toRight>{`${translate("DATE")}: ${formatDate(
                b.date
              )}`}</Name>
            </InfoWrapper>
          </MobileRow>

          <MobileRow>
            <Label>{`${translate("GROSS_VALUE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.grossValue, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>
          <MobileRow>
            <Label>{`${translate("NET_VALUE")} (R$)`}</Label>
            <PrincipalValue>
              {formatNumber(b.netValue, { digits: 2 })}
            </PrincipalValue>
          </MobileRow>

          <Separator />

          <MobileGrid>
            <GridInfo>
              <Label>{translate("TYPE")}</Label>
              <SecondaryValue>{b.type}</SecondaryValue>
            </GridInfo>
            <GridInfoLeftBorder>
              <Label>{`${translate("INCOME_TAX")} (R$)`}</Label>
              <SecondaryValue>
                {formatNumber(b.incomeTax, { digits: 2 })}
              </SecondaryValue>
            </GridInfoLeftBorder>
            <GridInfoLeftBorder>
              <Label>{`${translate("IOF")} (R$)`}</Label>
              <SecondaryValue>
                {formatNumber(b.iof, { digits: 2 })}
              </SecondaryValue>
            </GridInfoLeftBorder>
          </MobileGrid>
        </Card>
      </AssetRow>
    </AssetWrapper>
  ));
};

const buildTopInfo = (totaltransactions, loading) => {
  const incomeTax = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totaltransactions && totaltransactions.incomeTax, {
      digits: 2
    })
  );

  const iof = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totaltransactions && totaltransactions.iof, { digits: 2 })
  );

  const grossValue = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totaltransactions && totaltransactions.grossValue, {
      digits: 2
    })
  );

  const netValue = loading ? (
    <ShimmerLoading />
  ) : (
    formatNumber(totaltransactions && totaltransactions.netValue, { digits: 2 })
  );

  return (
    <TopHeader>
      <TitleHeader>{translate("TOTAL")}</TitleHeader>
      <HeaderData>
        <HeaderInfo>
          <HeaderLabel>{`${translate("INCOME_TAX")} (R$`}</HeaderLabel>
          <HeaderText>{incomeTax}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("IOF")} (R$)`}</HeaderLabel>
          <HeaderText>{iof}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("GROSS_VALUE")} (R$)`}</HeaderLabel>
          <HeaderText>{grossValue}</HeaderText>
        </HeaderInfo>
        <HeaderInfo>
          <HeaderLabel>{`${translate("NET_VALUE")} (R$)`}</HeaderLabel>
          <HeaderText>{netValue}</HeaderText>
        </HeaderInfo>
      </HeaderData>
    </TopHeader>
  );
};

function Mobile({
  fundsIncomeTransactions,
  isEmpty,
  activePage,
  totalCount,
  loading,
  onFilter,
  onChangePage
}) {
  const totaltransactions = generateSum(fundsIncomeTransactions);
  return (
    <Fragment>
      {!isEmpty && <Filter search={onFilter} />}
      {isEmpty && (
        <DefaultContent
          data-test="Empty_Position"
          Icon={() => <Icon type="EmptyWallet" color={black30} />}
          primaryText={translate("NO_RECORDS")}
          secondaryTexts={[translate("NO_RECORD_WAS_FOUND")]}
        />
      )}
      {!isEmpty && (
        <Wrapper>
          {buildTopInfo(totaltransactions, loading)}
          {buildRows(fundsIncomeTransactions, loading)}
        </Wrapper>
      )}
      {!isEmpty && (
        <Pagination
          onChangePage={onChangePage}
          pageTotal={totalCount}
          activePage={activePage}
          pageSize={PAGE_SIZE}
        />
      )}
      {!isEmpty && !loading && (
        <LinkCardWrapper>
          <LinkCard
            iconType="FiLoader"
            to="/investments/overview?menu=movements&from=funds"
            anchorText={translate("CLICK_TO_VIEW_FUNDS")}
            versionText={translate("WISH_TO_SEE_SIMPLIFIED_VERSION")}
            dataTest="LinkToSimplified"
            noSpan
            withUnderline
            fontSize={14}
          />
        </LinkCardWrapper>
      )}
    </Fragment>
  );
}

Mobile.defaultProps = {
  fundsIncomeTransactions: [],
  totalCount: 0
};

Mobile.propTypes = {
  fundsIncomeTransactions: arrayOf(
    shape({
      assetName: string,
      type: string,
      date: string,
      grossValue: number,
      netValue: number,
      incomeTax: number,
      iof: number
    })
  ),
  loading: bool,
  isEmpty: bool,
  totalCount: number,
  onFilter: func.isRequired,
  onChangePage: func.isRequired
};

export default Mobile;
