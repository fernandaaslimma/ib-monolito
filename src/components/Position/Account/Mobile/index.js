import { default as Mobile } from "./Mobile";

export default Mobile;
