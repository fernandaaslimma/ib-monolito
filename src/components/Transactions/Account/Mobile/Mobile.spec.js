import React from "react";
import { shallow } from "enzyme";
import Mobile from "./Mobile";

jest.mock("../../../../utils/i18n", () => ({
  translate: id => id,
  getDateFieldPlaceholderByLocale: () => "MM/DD/YYYY",
  getLanguage: () => {},
  isZhCN: () => false
}));

const cashAccounts = [
  {
    cashAccountTransactions: [
      {
        amount: 78351.22,
        date: "2018-01-03",
        type: "CRÉDITO"
      },
      {
        amount: 4923.16,
        date: "2018-01-08",
        type: "CRÉDITO"
      },
      {
        amount: -87879.5,
        date: "2018-01-09",
        type: "DÉBITO"
      },
      {
        amount: 88038.51,
        date: "2018-01-22",
        type: "CRÉDITO"
      },
      {
        amount: -91651.78,
        date: "2018-01-22",
        type: "DÉBITO"
      }
    ],
    userCashAccount: "308642-4",
    accountOpenBalance: {
      date: "2018-04-01",
      openBalance: 6903.2,
      closeBalance: 6903.2
    },
    accountCloseBalance: {
      date: "2018-05-04",
      openBalance: 52.6,
      closeBalance: 52.6
    }
  }
];

describe("Mobile component", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<Mobile cashAccounts={cashAccounts} loading={false} />)
    ).toMatchSnapshot();
  });

  it("should match snapshot with loading", () => {
    expect(
      shallow(<Mobile cashAccounts={cashAccounts} loading={true} />)
    ).toMatchSnapshot();
  });
});
