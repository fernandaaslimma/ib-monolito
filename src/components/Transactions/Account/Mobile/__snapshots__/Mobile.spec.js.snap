// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile component should match snapshot 1`] = `
<styled.div>
  <styled.section
    key="0"
  >
    <styled.div>
      <styled.h1>
        CHECKING_ACCOUNT 308642-4
      </styled.h1>
      <Card
        icon={null}
        iconColor={null}
        title={null}
        titleColor={null}
      >
        <styled.div>
          <styled.div>
            OPEN_BALANCE_ON 06/01/2010
          </styled.div>
          <styled.div>
            <styled.span
              red={false}
            >
              6.903,20
            </styled.span>
          </styled.div>
        </styled.div>
        <Styled(styled.div)
          noMargin={true}
        />
        <styled.div
          data-test="MobileGrid-0"
          key="MobileGrid_0"
        >
          <styled.div>
            <styled.div>
              DATE
            </styled.div>
            <styled.div>
              06/01/2010
            </styled.div>
          </styled.div>
          <Styled(styled.div)>
            <styled.div>
              TYPE
            </styled.div>
            <styled.div>
              CRÉDITO
            </styled.div>
          </Styled(styled.div)>
          <Styled(styled.div)>
            <styled.div>
              VALUE (R$)
            </styled.div>
            <styled.div>
              78.351,22
               
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
        <styled.div
          data-test="MobileGrid-1"
          key="MobileGrid_1"
        >
          <styled.div>
            <styled.div>
              DATE
            </styled.div>
            <styled.div>
              06/01/2010
            </styled.div>
          </styled.div>
          <Styled(styled.div)>
            <styled.div>
              TYPE
            </styled.div>
            <styled.div>
              CRÉDITO
            </styled.div>
          </Styled(styled.div)>
          <Styled(styled.div)>
            <styled.div>
              VALUE (R$)
            </styled.div>
            <styled.div>
              4.923,16
               
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
        <styled.div
          data-test="MobileGrid-2"
          key="MobileGrid_2"
        >
          <styled.div>
            <styled.div>
              DATE
            </styled.div>
            <styled.div>
              06/01/2010
            </styled.div>
          </styled.div>
          <Styled(styled.div)>
            <styled.div>
              TYPE
            </styled.div>
            <styled.div>
              DÉBITO
            </styled.div>
          </Styled(styled.div)>
          <Styled(styled.div)>
            <styled.div>
              VALUE (R$)
            </styled.div>
            <styled.div>
              -87.879,50
               
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
        <styled.div
          data-test="MobileGrid-3"
          key="MobileGrid_3"
        >
          <styled.div>
            <styled.div>
              DATE
            </styled.div>
            <styled.div>
              06/01/2010
            </styled.div>
          </styled.div>
          <Styled(styled.div)>
            <styled.div>
              TYPE
            </styled.div>
            <styled.div>
              CRÉDITO
            </styled.div>
          </Styled(styled.div)>
          <Styled(styled.div)>
            <styled.div>
              VALUE (R$)
            </styled.div>
            <styled.div>
              88.038,51
               
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
        <styled.div
          data-test="MobileGrid-4"
          key="MobileGrid_4"
        >
          <styled.div>
            <styled.div>
              DATE
            </styled.div>
            <styled.div>
              06/01/2010
            </styled.div>
          </styled.div>
          <Styled(styled.div)>
            <styled.div>
              TYPE
            </styled.div>
            <styled.div>
              DÉBITO
            </styled.div>
          </Styled(styled.div)>
          <Styled(styled.div)>
            <styled.div>
              VALUE (R$)
            </styled.div>
            <styled.div>
              -91.651,78
               
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
        <Styled(styled.div)
          noMargin={true}
        />
        <br />
        <styled.div>
          <styled.div>
            CLOSE_BALANCE_ON 06/01/2010
          </styled.div>
          <styled.div>
            <styled.span
              red={false}
            >
              52,60
            </styled.span>
          </styled.div>
        </styled.div>
      </Card>
    </styled.div>
  </styled.section>
  ;
</styled.div>
`;

exports[`Mobile component should match snapshot with loading 1`] = `
<styled.div>
  <styled.section>
    <styled.div>
      <Card
        icon={null}
        iconColor={null}
        title={null}
        titleColor={null}
      >
        <styled.div>
          <styled.div>
            OPEN_BALANCE_ON:
          </styled.div>
          <ShimmerLoading
            darker={false}
            index={0}
            inverse={false}
          />
        </styled.div>
        <Styled(styled.div)
          noMargin={true}
        />
        <styled.div>
          <styled.div>
            <styled.div>
              DATE
            </styled.div>
            <styled.div>
              <ShimmerLoading
                darker={false}
                index={0}
                inverse={false}
              />
            </styled.div>
          </styled.div>
          <Styled(styled.div)>
            <styled.div>
               
              TYPE
            </styled.div>
            <styled.div>
              <ShimmerLoading
                darker={false}
                index={0}
                inverse={false}
              />
            </styled.div>
          </Styled(styled.div)>
          <Styled(styled.div)>
            <styled.div>
               
              VALUE (R$)
            </styled.div>
            <styled.div>
              <ShimmerLoading
                darker={false}
                index={0}
                inverse={false}
              />
            </styled.div>
          </Styled(styled.div)>
        </styled.div>
        <Styled(styled.div)
          noMargin={true}
        />
        <styled.div>
          <styled.div>
            CLOSE_BALANCE_ON:
          </styled.div>
          <ShimmerLoading
            darker={false}
            index={0}
            inverse={false}
          />
        </styled.div>
      </Card>
    </styled.div>
  </styled.section>
  ;
</styled.div>
`;
